File: Quickstart.java
// QuanticData API quickstart — Java 11+ (single-file source mode, no build tool).
//   QUANTICDATA_API_KEY=qd_live_... java Quickstart.java
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.time.Duration;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class Quickstart {

    private static final String BASE = "https://api.quanticdata.io/v1";
    private static final HttpClient CLIENT =
            HttpClient.newBuilder().connectTimeout(Duration.ofSeconds(20)).build();

    /** POST a JSON body and return the raw response text. */
    private static String call(String path, String json) throws Exception {
        String key = System.getenv("QUANTICDATA_API_KEY");
        HttpRequest req = HttpRequest.newBuilder(URI.create(BASE + path))
                .header("Authorization", "Bearer " + key)
                .header("Content-Type", "application/json")
                .timeout(Duration.ofSeconds(120))
                .POST(HttpRequest.BodyPublishers.ofString(json))
                .build();

        HttpResponse<String> res = CLIENT.send(req, HttpResponse.BodyHandlers.ofString());
        if (res.statusCode() >= 400) {
            throw new IllegalStateException(path + " failed (" + res.statusCode() + "): " + res.body());
        }
        return res.body();
    }

    /**
     * The JDK ships no JSON parser, and this file deliberately has no dependencies —
     * so we pull the two fields we print with a regex. In real code use Jackson or Gson
     * and map the { type, message, payload } envelope to a record.
     */
    private static String field(String json, String name) {
        Matcher m = Pattern.compile("\"" + name + "\"\\s*:\\s*\"((?:[^\"\\\\]|\\\\.)*)\"").matcher(json);
        return m.find() ? m.group(1).replace("\\n", "\n").replace("\\\"", "\"") : "";
    }

    public static void main(String[] args) throws Exception {
        if (System.getenv("QUANTICDATA_API_KEY") == null) {
            System.err.println("set QUANTICDATA_API_KEY — https://app.quanticdata.io/register");
            System.exit(1);
        }

        String page = call("/scrape", "{\"url\":\"https://example.com\",\"format\":\"markdown\"}");
        String markdown = field(page, "markdown");
        System.out.println("--- markdown -------------------------------------------------");
        System.out.println(markdown.substring(0, Math.min(400, markdown.length())));
        System.out.println("title: " + field(page, "title"));

        String serp = call("/serp", "{\"query\":\"web scraping api\",\"country\":\"us\",\"num\":5}");
        System.out.println();
        System.out.println("--- organic (raw JSON, first 600 chars) ----------------------");
        System.out.println(serp.substring(0, Math.min(600, serp.length())));
    }
}

File: Quickstart.cs
// QuanticData API quickstart — C# / .NET 8 (System.Net.Http + System.Text.Json).
//   QUANTICDATA_API_KEY=qd_live_... dotnet run
using System.Text;
using System.Text.Json;

const string Base = "https://api.quanticdata.io/v1";

var key = Environment.GetEnvironmentVariable("QUANTICDATA_API_KEY");
if (string.IsNullOrWhiteSpace(key))
{
    Console.Error.WriteLine("set QUANTICDATA_API_KEY — https://app.quanticdata.io/register");
    return 1;
}

using var http = new HttpClient { Timeout = TimeSpan.FromSeconds(120) };
http.DefaultRequestHeaders.Add("Authorization", $"Bearer {key}");

// POST to the API and unwrap the { type, message, payload } envelope.
async Task<JsonElement> CallAsync(string path, object body)
{
    var json = JsonSerializer.Serialize(body);
    using var content = new StringContent(json, Encoding.UTF8, "application/json");
    using var res = await http.PostAsync(Base + path, content);
    var text = await res.Content.ReadAsStringAsync();

    using var doc = JsonDocument.Parse(text);
    var root = doc.RootElement;
    var type = root.TryGetProperty("type", out var t) ? t.GetString() : null;
    if (!res.IsSuccessStatusCode || type == "error")
    {
        var msg = root.TryGetProperty("message", out var m) ? m.GetString() : text;
        throw new InvalidOperationException($"{path} failed ({(int)res.StatusCode}): {msg}");
    }
    return root.GetProperty("payload").Clone();
}

var page = await CallAsync("/scrape", new { url = "https://example.com", format = "markdown" });
var markdown = page.TryGetProperty("markdown", out var md) ? md.GetString() ?? "" : "";
Console.WriteLine("--- markdown -------------------------------------------------");
Console.WriteLine(markdown[..Math.Min(400, markdown.Length)]);

var serp = await CallAsync("/serp", new { query = "web scraping api", country = "us", num = 5 });
Console.WriteLine();
Console.WriteLine("--- organic --------------------------------------------------");
if (serp.TryGetProperty("organic", out var organic))
{
    foreach (var row in organic.EnumerateArray())
    {
        var position = row.TryGetProperty("position", out var p) ? p.ToString() : "?";
        var title = row.TryGetProperty("title", out var ti) ? ti.GetString() : "";
        var link = row.TryGetProperty("link", out var l) ? l.GetString() : "";
        Console.WriteLine($"{position,2}. {title}\n    {link}");
    }
}

return 0;
